commands/list/nodepools/command.go

```diff
@@ -222,6 +217,33 @@ func getOutput(nps []*models.V5GetNodePoolsResponseItems, outputFormat string) (
 		return string(outputBytes), nil
 	}
 
+	var output string
+	var err error
+	np := nps[0]
+
+	if np.NodeSpec.Aws != nil && np.NodeSpec.Azure == nil {
+		output, err = getOutputAWS(nps)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	} else if np.NodeSpec.Azure != nil && np.NodeSpec.Aws == nil {
+		output, err = getOutputAzure(nps)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	} else {
+		return "", microerror.Mask(errors.ClusterDoesNotSupportNodePoolsError)
```

Comment:
I'm just not sure what to display if we cannot figure out which provider is this (which would mean that there is an issue in the API). An error? Or maybe only show fields that are common across all providers?

Reply:
`fetchNodepools()` already does this check (and returns same error). I think this is enough here.